lib/server.js

```diff
@@ -243,6 +242,123 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
+        '/consensus/continuity2017/voters/' + req.params.voterId;
+      const supportedVoteTopics = ['Events', 'RollCall'];
+      const voteTopic = req.params.voteTopic;
+      if(!supportedVoteTopics.includes(voteTopic)) {
+        return next(new BedrockError(
+          'Unsupported vote topic.', 'NotSupportedError', {
+            httpStatusCode: 400,
+            public: true,
+            voteTopic,
+            supportedVoteTopics,
+          }));
+      }
+      async.auto({
+        voteHash: callback =>
+          brLedger.consensus._hasher(req.body, callback),
+        voter: callback => storage.voters.get({voterId}, callback),
+        ledgerNode: ['voter', (results, callback) =>
+          brLedger.get(null, results.voter.ledgerNodeId, callback)],
+        addVote: ['voteHash', 'ledgerNode', (results, callback) =>
+          storage.votes.add(results.ledgerNode.id, voteTopic, req.body, {
+            meta: {voteHash: results.voteHash}
+          }, callback)]
+      }, err => {
+        if(err && !(database.isDuplicateError(err) ||
+          err.name === 'DuplicateError')) {
+          return next(err);
+        }
+        res.status(200).end();
+      });
+    });
+  docs.annotate.post(routes.votes, {
+    description: 'Add a new vote',
+    schema: 'continuity.vote',
+    securedBy: ['null'],
+    responses: {
+      200: 'Vote was accepted for writing. HTTP Location header ' +
+        'contains URL of accepted event.',
+      400: 'Request failed due to malformed request.',
+    }
+  });
+
+  // Add a new manifest
+  app.post(
+    routes.manifests, brRest.when.prefers.ld, validate('continuity.manifest'),
+    (req, res, next) => {
+      const voterId = config.server.baseUri +
+        '/consensus/continuity2017/voters/' + req.params.voterId;
+      const manifest = req.body;
+      return async.auto({
+        validateHash: callback => _validateManifestHash(manifest, callback),
+        voter: ['validateHash', (results, callback) =>
+          storage.voters.get({voterId}, callback)],
+        ledgerNode: ['voter', (results, callback) =>
+          brLedger.get(null, results.voter.ledgerNodeId, callback)],
+        checkExists: ['ledgerNode', (results, callback) =>
+          storage.manifests.exists(
+            results.ledgerNode.id, manifest.id, (err, result) => {
+              if(err) {
+                return callback(err);
+              }
+              if(result) {
+                return callback(new BedrockError(
+                  'The manifest already exists.', 'DuplicateError', {
+                    manifest: manifest.id
+                  }, err));
+              }
+              callback();
+            })],
+        checkMajority: ['checkExists', (results, callback) =>
+          election.checkMajority(manifest, results.ledgerNode, callback)],
+        checkItems: ['checkMajority', (results, callback) => {
+          // FIXME: is it necessary to check items if the type is `Events`?
```

Comment:
Yes, we need to check for event hashes, not vote hashes, in the case where the topic is `Events`.

Reply:
@dlongley to clarify the question, I *believe* it is safe to assume that we already have all the `events`?

Reply:
That's based on a presumption that the client isn't misbehaving -- which we shouldn't assume.

Reply:
Our client won't intentionally misbehave -- but we should not assume that someone else won't write one that will (or that ours won't have bugs).